File: Daily Test/Python/Bar Graph Pattern - Custom Width 7.9.2025.py
N = int(input())
height = list(map(int, input().split()))
X = int(input())

maxH = max(height)
tot = []

for i in range(N):
    cnt = 0
    while cnt < X:
        bar = '.' * (maxH - height[i]) + '*' * height[i]
        tot.append(bar)
        cnt += 1

for i in range(maxH):
    for word in tot:
        print(word[i], end=' ')
    print()

File: Daily Test/Java/Bar Graph Pattern - Custom Width 7.9.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] height = new int[N];
        for (int i = 0; i < N; i++) height[i] = sc.nextInt();
        int X = sc.nextInt();

        int maxH = Arrays.stream(height).max().getAsInt();
        ArrayList<String> tot = new ArrayList<>();

        for (int i = 0; i < N; i++) {
            for (int c = 0; c < X; c++) {
                StringBuilder bar = new StringBuilder();
                for (int j = 0; j < maxH - height[i]; j++) bar.append(".");
                for (int j = 0; j < height[i]; j++) bar.append("*");
                tot.add(bar.toString());
            }
        }

        for (int i = 0; i < maxH; i++) {
            for (String word : tot) {
                System.out.print(word.charAt(i) + " ");
            }
            System.out.println();
        }
    }
}